services/compression/engine.ts

```
@@ -0,0 +1,572 @@
+  try {
+    // Try standard US format: MM/DD/YYYY
+    const parts = dateStr.split("/");
+    if (parts.length === 3) {
+      const month = parseInt(parts[0], 10);
+      const day = parseInt(parts[1], 10);
+      const year = parseInt(parts[2], 10);
+
+      // Validate ranges by attempting to construct a Date object
+      const date = new Date(year, month - 1, day);
+      if (
+        isNaN(date.getTime()) ||
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)